### ISSUE TITLE
Make wipe optional inlullabot/drainpipe/scaffold/github/actions/pantheon/review/action.yml

### ISSUE THREAD

**Issue by tess-ten7** [Lullabot/drainpipe#357]

**Comment by justafish** [comment]
@tess-ten7 👋 do you have any examples of scenarios where this is counter productive? In general we try to keep things as "repeatable" as possible i.e. a PR rebuild would always start things from the same state and configure everything as necessary, rather relying on the database to be in a specific special state as this creates an environment that won't necessarily match what happens on deployment

**Comment by tess-ten7** [comment]
@justafish There are two use cases which come to mind:

1. The upstream database/files are rather large, drastically increasing our Github action costs while those sync.
2. The feature being worked on in the PR requires special content to be built to support it, which cannot be implemented on live at this time.

Making the wipe optional allows us to handle both of these issues by deferring the need to wipe and resync to the developer.

**Comment by generalludd** [comment]
I agree with @tess-ten7 development environments exist to test not only code but code and content. For paying clients, time spent waiting for a larger site to copy files and database every build is costly especially when we have to restore a working db or files with sample content for demonstration purposes. We either have to absorb the costs in development time or we have to pass them on to the client.

**Comment by justafish** [comment]
Thanks for your feedback :+1:
Do you have any numbers around the time it's taking to copy a database/set of files of x size?

For use case 2. we'd tend to use https://www.drupal.org/project/default_content/ and then remove it if necessary before merge

**Comment by generalludd** [comment]
On a moderately large site (say one with a 50MB or larger DB, it takes as much as 10 additional minutes to restore a database after a build (which also takes about 3-5 more minutes per build to copy a new db. So in the end we're spending as much as 20 minutes on a build and we can't step away from the work because we need to reload the db so we can continue to show a client the changes we're proposing. This is most commonly an issue when we're working on a large epic that involves, say, a new functionality on the site.

**Comment by generalludd** [comment]
@justafish Here's a patch we use on one of our sites. It could just be a patch here that could be offered until some option could be to incorporate. You have a marvelous product and I understand this is a product mainly designed for Lullabot's workflows, but this could be a useful option (possibly even in your own shop)
[drainpipe-skip-wipe-357.patch](https://github.com/user-attachments/files/15957355/drainpipe-skip-wipe-357.patch)

**Comment by mrdavidburns** [comment]
This feature would be a useful option for our support clients as well. We recently had a project where we enabled this feature but the client told us to disable it because of how many GitHub Actions minutes were being used. cc @YesCT @DBNubs

**Comment by YesCT** [comment]
Yeah, we need the job to be super fast for adding the job to the customer that is sensitive to their GitHub minutes usage. Looping in  @leonel-lullabot also who is doing some investigation on where the time is being spent in the PantheonReviewApps / multidev deploys.

**Comment by mrdavidburns** [comment]
@mmenavas Evaluate if [terminus env:wipe](https://docs.pantheon.io/terminus/commands/env-wipe) is any faster than [terminus multidev:delete](https://docs.pantheon.io/terminus/commands/multidev-delete)

**Comment by mrdavidburns** [comment]
We've recently merged a configuration option that allows Pantheon MultiDev to be done with a manual GitHub Action.
see: #535

**Comment by YesCT** [comment]
Different, but related idea for config: instead of wiping a multidev env, delete it and make it newly. I think that would be faster than wiping and remaking it. Made a new feature request: https://github.com/Lullabot/drainpipe/issues/796

**Comment by mrdavidburns** [comment]
@deviantintegral and I discussed this during the Drainpipe sync yesterday. Both of us have a recollection that this was tried on existing projects and more frequently than not the GitHub Action would fail. Not sure if it was an error or a timeout. Switching to wipe reduced the frequency that this failure would occur. 

We can try again and figure out exactly why it failed but after discussing with @deviantintegral I'm less optimistic that this will be the path forward.

**Comment by generalludd** [comment]
We have been using this patch on over ten actively developed sites for nearly two years and have not had any issues. It could be that there be an option in drainpipe configs to turn wiping on or off. Then if users have trouble with builds, they could turn it off.

**Comment by BlancaEsqueda-Empire** [comment]
Thanks @generalludd we've been using the patch as well.
After a composer update the patch wasn't applying, this is the one updated:

[drainpipe-skip-wipe-357-3.patch](https://github.com/user-attachments/files/25721747/drainpipe-skip-wipe-357-3.patch)
